packages/openapi-fetch/test/common/response.test.ts

```diff
@@ -20,22 +20,38 @@ describe("response", () => {
       // 2. assert data is not undefined inside condition block
       if (result.data) {
         assertType<NonNullable<Resource[]>>(result.data);
-        assertType<undefined>(result.error);
+        // @ts-expect-error FIXME: This is a limitation within Typescript
+        assertType<never>(result.error);
       }
       // 2b. inverse should work, too
       if (!result.error) {
         assertType<NonNullable<Resource[]>>(result.data);
-        assertType<undefined>(result.error);
+        assertType<never>(result.error);
+      }
+
+      if (result.status === 200) {
+        assertType<NonNullable<Resource[]>>(result.data);
+        assertType<never>(result.error);
```

Comment:
These are good tests! I would want to see the other common codes tested here, too—`201`, and `400`–`405`, just for completeness (and if they’re not in the schema, we should add them!)